keras_nlp/metrics/perplexity_test.py

```diff
@@ -0,0 +1,148 @@
+# Copyright 2022 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for Perplexity."""
+
+import tensorflow as tf
+
+from keras_nlp.metrics import Perplexity
+
+
+class PerplexityTest(tf.test.TestCase):
+    @classmethod
```

Comment:
Curious: why are you using this classmethod `setUpClass` instead of `setUp(self)`?

Also I would prefer setting the value of x, y in separate test cases - the disadvantage is we are copying code around, but the advantage is that the test case is more clear on its purposes and data.

Reply:
`setUp()` runs before every unit test, whereas, `setUpClass()` runs only once before all the UTs.

For some reason, even though I've set the random seed, `setUp()` gives different outputs for every test. That's why I used `setUpClass()`.